### PR TITLE
Added MaxActions Limit Check

### LINKED ISSUE #3695: [iOS][HostConfig] [Number of Actions Setting Is Not Supported]
# Platform

* iOS

# Version of SDK

1.2.5

# Details

HostConfig's Max Actions setting is not supported in iOS

### PR DESCRIPTION
**Pull request by nesalang**
## Related Issue
* fixes #3695

## Description
* Added MaxActions limit check
*  Replaced multiple GetActions() call that return a structure by value

## How Verified
* verified it by changing values for max actions in HostConfig.

###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/microsoft/AdaptiveCards/pull/3696)

### CODE REVIEW

File: source/ios/AdaptiveCards/AdaptiveCards/AdaptiveCards/ACRActionSetRenderer.mm [hunk 1]
```diff
@@ -76,11 +77,14 @@ - (UIView *)renderButtons:(ACRView *)rootView
...
     [superview addArrangedSubview:containingView];
-    float accumulatedWidth = 0, accumulatedHeight = 0, spacing = [config getHostConfig] -> GetActions().buttonSpacing, maxWidth = 0, maxHeight = 0;
+    float accumulatedWidth = 0, accumulatedHeight = 0, spacing = adaptiveActionConfig.buttonSpacing, maxWidth = 0, maxHeight = 0;
     if (elems.empty()) {
         return containingView;
     }
-    for (const auto &elem : elems) {
+
+    unsigned long uMaxActionsToRender = MIN(adaptiveActionConfig.maxActions, elems.size());
+    for (auto i = 0; i < uMaxActionsToRender; i++) {
```
Comment (RebeccaAnne) [inline source/ios/AdaptiveCards/AdaptiveCards/AdaptiveCards/ACRActionSetRenderer.mm]: >for (auto i = 0; i < uMaxActionsToRender; i++) [](start = 4, length = 46)

Should we add a warning if we have more than the max actions?  Just checked and we do in UWP.

Comment (paulcam206) [reply source/ios/AdaptiveCards/AdaptiveCards/AdaptiveCards/ACRActionSetRenderer.mm]: agree with @RebeccaAnne here :)

---
In reply to: [363433129](https://github.com/microsoft/AdaptiveCards/pull/3696#discussion_r363433129) [](ancestors = 363433129)

Comment (shalinijoshi19) [reply source/ios/AdaptiveCards/AdaptiveCards/AdaptiveCards/ACRActionSetRenderer.mm]: @nesalang we should also have unit tests for this.

**Comment by VikrantSingh01** [comment]
@nesalang @shalinijoshi19 : Hoping this will be part of next/ 1.2.6 drop ?

**Comment by nesalang** [comment]
yes

---
In reply to: [572468215](https://github.com/microsoft/AdaptiveCards/pull/3696#issuecomment-572468215) [](ancestors = 572468215)

**Pull request merged**